### ISSUE TITLE
Need practical documentation on using Adapters

### ISSUE THREAD

**Issue by Sammaye** [select2/select2#3166]
There is clear documentation on what builds up an adapter but there seems to be no clear documentation on how to actually use that adapter with select2, not even an example.

**Comment by arcooverbeek** [comment]
:+1:

**Comment by kevin-brown** [comment]
Useful links for setting up adapters and using them

- Documentation: https://github.com/select2/select2/issues/3116
- Example: http://stackoverflow.com/a/30631849/359284

**Comment by marcodelpercio** [comment]
Hi,

I just wanted to share my personal experience. It personally took me quite a long time to find a working example of how to create a custom dataAdapter. The documentations is definitely not clear enough. There're not many examples online. The only example contained in the 4.0 announcement has an error (in fact it's missing ");" at the end)   I've also just spent hours looking for documentation/examples on how to set up InfiniteScroll with version 4.x. I'm quitting and going back to version 3.5. Quicker and simpler

**Comment by bojanv91** [comment]
Just stumbled upon this issue. I was also having hard time to understand "how to use" the adapters and decorators, as I could not find working examples online. 

So, after I figured it out (mostly by looking at how select2 internally uses the adapters feature) I wrote a short blog post as a note/reminder. I hope someone else might find it useful too, so I'll post the link here:
https://bojanv91.github.io/posts/2017/10/extending-select2-with-adapters-and-decorators
